webapp/src/components/Dialogs/ManageProjectDialog.jsx

```diff
@@ -9,45 +9,71 @@ import Dialog from '@material-ui/core/Dialog';
+import FormControl from '@material-ui/core/FormControl';
 import HighlightOff from '@material-ui/icons/HighlightOff';
 import IconButton from '@material-ui/core/IconButton';
 import List from '@material-ui/core/List';
 import ListItem from '@material-ui/core/ListItem';
 import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
 import ListItemText from '@material-ui/core/ListItemText';
-import TextField from '@material-ui/core/TextField';
+import MenuItem from '@material-ui/core/MenuItem';
+import OutlinedInput from '@material-ui/core/OutlinedInput';
 import Typography from '@material-ui/core/Typography';
 
-import { projectsApi } from '../../services/contaxy-api';
+import { Select } from '@material-ui/core';
+import { projectsApi, usersApi } from '../../services/contaxy-api';
 import { useProjectMembers } from '../../services/api-hooks';
 import GlobalStateContainer from '../../app/store';
+import UserSearch from '../UserSearch';
 import showStandardSnackbar from '../../app/showStandardSnackbar';
 
 function ManageProjectDialog(props) {
   const { className, project, onClose } = props;
   const globalStateContainer = GlobalStateContainer.useContainer();
   const { loadProjects } = globalStateContainer;
-  const [userToAdd, setUserToAdd] = useState(''); // set to empty object so that material-ui knows that it is a controlled input
+  const [users, setUsers] = useState([]);
+  const [selectedUser, setSelectedUser] = useState(''); // set to empty object so that material-ui knows that it is a controlled input
+  const [permission, setPermission] = useState('read');
   const [projectMembers, reloadProjectMembers] = useProjectMembers(project.id);
 
-  const handleSelectUser = (e) => {
-    setUserToAdd(e.target.value);
+  const onUserSelect = (user) => {
+    setSelectedUser(user);
   };
 
+  async function requestUsers(projectMembers) {
+    let response;
+    try {
+      response = await usersApi.listUsers();
+      const userList = response.filter(
+        (a) => !projectMembers.some((b) => a.id === b.id)
+      );
+      setUsers(userList);
+    } catch (e) {
+      setUsers([]);
+    }
+  }
+
   const handleAddUser = async () => {
     try {
-      await projectsApi.addProjectMember(project.id, userToAdd);
+      await projectsApi.addProjectMember(project.id, selectedUser.id, {
+        accessLevel: permission,
+      });
       showStandardSnackbar(
-        `Added user '${userToAdd}' to project '${project.id}'.`
+        `Added user '${selectedUser.username}' to project '${project.id} with permission '${permission}''.`
       );
+      onUserSelect('');
       reloadProjectMembers();
     } catch (e) {
       showStandardSnackbar(
-        `Could not add user '${userToAdd}' to project '${project.id}'.`
+        `Could not add user '${selectedUser.username}' to project '${project.id}'.`
       );
     }
   };
 
+  useEffect(() => {
+    requestUsers(projectMembers);
+  }, [setUsers, projectMembers, selectedUser]);
```

Comment:
I think adding selectedUser to the dependency list of useEffect will cause a reload of all ML Lab users on each selection change of the dropdown. It should be safe to delete and reduce the number of calls to the api/users endpoint. (To further optimize, we could also only call the api/users endpoint only once when this component is mounted and cache the result. However, I don't think we need to optimize that now already)